src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/LogicalExpression.java

```diff
@@ -0,0 +1,95 @@
+package io.stargate.sgv2.jsonapi.api.model.command.clause.filter;
+
+import io.stargate.sgv2.jsonapi.config.constants.DocumentConstants;
+import java.util.*;
+
+/**
+ * This object represents conditions based for a json path (node) that need to be tested Spec says
+ * you can do this, compare equals to a literal {"username" : "aaron"} This is a shortcut for
+ * {"username" : {"$eq" : "aaron"}} In here we expand the shortcut into a canonical long form, so it
+ * is all the same.
+ */
+public class LogicalExpression {
+
+  public enum LogicalOperator {
+    AND("$and"),
+    OR("$or");
+    private String operator;
+
+    LogicalOperator(String operator) {
+      this.operator = operator;
+    }
+
+    public String getOperator() {
+      return operator;
+    }
+  }
+
+  private final LogicalOperator logicalRelation;
+  private int totalComparisonExpressionCount;
+  private int totalIdComparisonExpressionCount;
+
+  public List<LogicalExpression> logicalExpressions;
+  public List<ComparisonExpression> comparisonExpressions;
+
+  private LogicalExpression(
+      LogicalOperator logicalRelation,
+      int totalComparisonExpressionCount,
+      List<LogicalExpression> logicalExpressions,
+      List<ComparisonExpression> comparisonExpression) {
+    this.logicalRelation = logicalRelation;
+    this.totalComparisonExpressionCount = totalComparisonExpressionCount;
+    this.logicalExpressions = logicalExpressions;
+    this.comparisonExpressions = comparisonExpression;
+  }
+
+  public static LogicalExpression and() {
+    return new LogicalExpression(LogicalOperator.AND, 0, new ArrayList<>(), new ArrayList<>());
+  }
+
+  public static LogicalExpression or() {
+    return new LogicalExpression(LogicalOperator.OR, 0, new ArrayList<>(), new ArrayList<>());
+  }
+
+  public void addLogicalExpression(LogicalExpression logicalExpression) {
+    // skip empty logic expression
+    if (logicalExpression.logicalExpressions.isEmpty()
```

Comment:
Maybe add `isEmpty()` method in this class so there's no need to reference internal fields from here?

So something like

```
public boolean isEmpty() {
   return logicalExpressions.isEmpty();
}
```